src/spaces/continuous.jl

```diff
@@ -410,3 +417,34 @@ function Base.iterate(iter::PairIterator, i = 1)
     id1, id2 = p
     return (iter.agents[id1], iter.agents[id2]), i + 1
 end
+
+
+#######################################################################################
+# %% Spatial fields
+#######################################################################################
+"""
+    get_spatial_property(pos::NTuple{D, Float64}, property::AbstractArray, model::ABM)
+Convert the continuous agent position into an appropriate `index` of `property`, which
+represents some discretization of a spatial field over a [`ContinuousSpace`](@ref).
+Then, return `property[index]`.
+
+The dimensionality of `property` and the continuous space must match.
+"""
+function get_spatial_property(pos, property::AbstractArray, model::ABM)
+    spacesize = model.space.extent
+    propertysize = size(property)
+    idxs = _convert_pos_to_cell(pos, spacesize, propertysize)
+    return property[idxs]
+end
+
+function _convert_pos_to_cell(pos, spacesize, propertysize)
+    εs = spacesize ./ propertysize
+    idxs = floor.(Int, pos ./ εs) .+ 1
+    return CartesianIndex(idxs)
+end
+
+"""
+    get_spatial_property(pos::NTuple{D, Float64}, property::Any, model::ABM)
```

Comment:
Should `property` really be `Any` here, or `Function`?

Reply:
`Any`. Function-like objects are a huge staple of Julia, and it is not always the case that the creator would have subtyped them as `Function`. And in some cases this isn't even possible.